pvmismatch/pvmismatch_lib/pvmodule.py

```diff
@@ -484,18 +498,48 @@ def calcMod(self):
                     Isub, Vsub = self.pvconst.calcParallel(
                         Iparallel, Vparallel, Vparallel.max(), Vparallel.min()
                     )
-            bypassed = Vsub < self.Vbypass
-            Vsub[bypassed] = self.Vbypass
+                    
+            # apply bypass diodes depending on the configuration passed
+            if type(self.Vbypass) == list:
+                # if only one value is passed in the list- assume only one 
+                # bypass diode  across the PV module
+                if len(self.Vbypass) == 1: 
+                    bypass_entire_module = True
+                    Vbypass_mod = self.Vbypass[0]
```

Comment:
I think that maybe a lot of this can be moved up to the constructor, and pre-configured before getting to this point.

* all of this doesn't have to be configured inside this loop every time, it wastes time and slows the model down, which is really undesirable IMO
* move some of this to the constructor, `__init__(self, ...)` for the `PVmodule`, for example, `bypass_entire_module` could be set _once_ in the constructor.
* it's probably more convenient to convert any scalars to a sequence in the constructor using duck typing similar to the example above. I usually call this method, `listify` and put it somewhere convenient, although you should probably make it a tuple since it should be immutable.

        # in pvmismatch_lib.utils or somewhere convenient
        def listify(sequence_or_scalar):
            """convert a scalar to sequence"""
            try:
                iter(sequence_or_scalar)
            except TypeError:
                sequence_or_scalar = [sequence_or_scalar]
            return sequence_or_scalar

        self.Vbypass = tuple(utils.listify(Vbypass))  # convert scalars and lists to a tuple